client/src/GameState.ts

```diff
@@ -174,43 +174,38 @@ export class GameState {
 
     public fromBoardState(boardState: BoardState): Array<Monomino> {
         const needRedraw = [];
-        if (this.playerId && this.playerId > 0) {
-            // rotate remote player0 board to local board
-            const ccRotations = 4 - ((this.playerId || 0) % 4);
-            const localBoard = boardState.map((row) => row.slice());
-            for (let row = 0; row < BOARD_SIZE; row++) {
-                for (let col = 0; col < BOARD_SIZE; col++) {
-                    const [newRow, newCol] = Tetromino.rotateCoords(
-                        [row, col],
-                        BOARD_SIZE,
-                        ccRotations
-                    );
-                    localBoard[newRow][newCol] = boardState[row][col];
-                    const monominoState = boardState[row][col];
+        const ccRotations = 4 - ((this.playerId || 0) % 4);
+
+        for (let row = 0; row < BOARD_SIZE; row++) {
+            for (let col = 0; col < BOARD_SIZE; col++) {
+                const [newRow, newCol] = Tetromino.rotateCoords(
+                    [row, col],
+                    BOARD_SIZE,
+                    ccRotations
+                );
+                // the future state of the monomino at [row, col]
+                let monominoState = boardState[row][col];
+                if (this.playerId && this.playerId > 0) {
+                    // rotate from player0 view
+                    monominoState = boardState[row][col];
                     if (monominoState) {
                         monominoState.position = [newRow, newCol];
                     }
                 }
-            }
-            boardState = localBoard;
-        }
-        for (let row = 0; row < BOARD_SIZE; row++) {
-            for (let col = 0; col < BOARD_SIZE; col++) {
-                const monominoState = boardState[row][col];
                 if (!monominoState) {
                     // erase existing monomino if it should now be null
-                    const existingMonomino = this.board[row][col];
+                    const existingMonomino = this.board[newRow][newCol];
                     if (existingMonomino) {
```

Comment:
Can just call this inline

Reply:
![image](https://user-images.githubusercontent.com/25297856/161192248-53ba29d5-209a-46ef-93a5-58dec98f9542.png)

typescript cannot tell if the element is a `false` or `Monomino` (or even `null` if without the `!` mark) if we re-index the array to get it. I would prefer inline too, but to not fight with typescript compiler I just used that var

Reply:
ah yep okay